programs/dibio.c

```diff
@@ -49,21 +49,25 @@
 static const size_t g_maxMemory = (sizeof(size_t) == 4) ? (2 GB - 64 MB) : ((size_t)(512 MB) << sizeof(size_t));
 
 #define NOISELENGTH 32
+#define MAX_SAMPLES_SIZE (2 GB) /* training dataset limited to 2GB */
 
 
 /*-*************************************
 *  Console display
 ***************************************/
+#define DISPLAY_LEVEL_DEFAULT 2
+static int g_displayLevel = DISPLAY_LEVEL_DEFAULT;
```

Comment:
I kind of remember that I tried to limit usage of global variables in this file.
The annoying part is that functions which display messages on console need to receive variables like `displayLevel`, which is a load on their parameter list.

As a side-effect, it's less easy now to "forget" synchronizing `displayLevel` between `zstd` modules, which all tend to have their own `g_displayLevel` that needs to be updated.

Not that it matters a lot, but reducing the amount of global variables felt somewhat cleaner, and now that the work is done, why reverting back to usage of global variables ?

Reply:
Yes, there are tradeoffs. I sense you want to do away with the global variable and pass displayLevel as parameter. I can make that change.